### ISSUE TITLE
"Getting started" for Windows:  `spress.cmd` error

### ISSUE THREAD

**Issue by tajmone** [spress/Spress#74]
In the "Getting started" page (http://spress.yosymfony.com/docs/getting-started/) it says:

```
5. Create a wrapping batch script (results in C:\bin\spress.cmd):

      C:\Users\username> cd C:\bin
      C:\bin> echo @php "%~dp0spress.phar" %* > spress.cmd
      C:\bin> exit
```
the line `echo @php "%~dp0spress.phar" %* > spress.cmd` is wrong: it should have `%%*` instead of `%*`, otherwise the variable gets expanded (to an empty string) during echo and is omitted in the final `spress.cmd` file — which prevents the `spress` cmd from passing on additional parameters!

Also, for the same reasons, it would be best to add an `%` before `%~dp0spress.phar`, so that even the script path isn't expanded (to the script literal path) during echo, which would allow to create a  `spress.cmd` file that works even if moved around after creation.

Therefore, the whole line should be changed to:

```
C:\bin> echo @php "%%~dp0spress.phar" %%* > spress.cmd
```

**Comment by yosymfony** [comment]
Thank you man. I don't use usually Window.

**Comment by tajmone** [comment]
Thanks to you, Spress is really a great software. As for Windows CMD—neither do I use it much. I stumbled across this issue while I was experimenting with the creation of a portable Spress package (with PHP binaries and all, no installation required), which forced me to look into it in detail.
Best regards!

**Comment by yosymfony** [comment]
A Spress package sounds interesting. Can you tell me more details about that?

**Comment by tajmone** [comment]
I'm working on a project for a computing-related shared documentation, possibly allowing to use Git version control. So I though of using Spress, because of Markdown ease, code highlighting, ecc.

On Windows  (as usual) even setting up something as simple as PHP might become a deterrant, so I wanted to make sure I could also provide a PHP binary distribution, in standalone format, that could be unzipped along an Spress project to be ready to go—and possibly, one that can work even if on the system other versions of PHP are present.

So far, I've managed to create an archive with PHP binaries (bare minimum configuration, and a few batch scripts) which allow to run an Spress pre-existing project, whithout need to install anything nor change the Windows registry—and worked on a machine with a pre-installed (older) version of PHP.

Linux and Max boxes shouldn't have any problems handling PHP projects. But I think that any Windows project should provide a ready-to-go binary to its potential users—instructions/tutorials end up scaring people even more, since on Windows everything tends to be more complex than needed.

So, basically thats it ... I'm laying the grounds for a sort of collaborative encyclopedia/collection of resources for programmers (sample codes included), based on GitHub and built on Spress. The idea is to have to whole project locally, instead of consulting online, in order to reduce the gap between using it and contributing to it—via Git, any changes and additions become, potentially, contribution to overall project.

It took me many tests with different flat CMS projects to end up finally choosing Spress—for various reasons, I consider it the best tool I've come across so far, at least for the project I have in mind.
Other were discarded for some reason or other.

**Comment by iola-lin** [comment]
Hi,
I installed spress on the windows 10.
But the following instruction was not work.
```shell
C:\bin> echo @php "%%~dp0spress.phar" %%* > spress.cmd
```
It would get the error here: 
```shell
Could not open input file: %~dp0spress.phar
```
Instead, I changed to the original statement, which is:
```shell
C:\bin> echo @php "%~dp0spress.phar" %* > spress.cmd
```
The `spress` command then worked.
![sp170107_203506](https://cloud.githubusercontent.com/assets/14487547/21741681/02525a2a-d519-11e6-9297-9b4f803f8d95.png)

Another thing is i do the above command on powershell.
The cmd would show like this:
![sp170107_203609](https://cloud.githubusercontent.com/assets/14487547/21741687/1c99b4c8-d519-11e6-8922-cd325d1b9b3e.png)

If there is any explanation, it would definitely help other windows users.
I'm not that familiar with those phar things to explain this situation.

**Comment by tajmone** [comment]
This has to do with the fact that in CMD/DOS double percent `%%` is used for variables, while single percent `%` is used for parameters.

 `%0` expands to the the pathname of the batch script itself. (`%0` is like an implicit parameter, even when no parameters are passed to the invoked script).

`%~dp0` expands to the current script's Drive letter + path (filename excluded). It contains two extended syntax modifiers:

- `%~d`: expands to the Drive letter
- `%~p`: expands to the Path only

which are joined together in `%~dp0`.

More info can be found here:

- http://ss64.com/nt/syntax-args.html
- http://ss64.com/nt/syntax-esc.html
- http://ss64.com/nt/set.html

**Comment by iola-lin** [comment]
@tajmone , thanks for your response. I will read through it.

**Comment by yosymfony** [comment]
I know I'm late with the response. The resources refered by @tajmone are very good
